modules/vault-cluster/main.tf

```diff
@@ -213,3 +214,27 @@ data "aws_iam_policy_document" "vault_s3" {
     ]
   }
 }
+
+resource "aws_iam_role_policy" "vault_aws_EC2_IAM_Auth" {
```

Comment:
Please use all lower case for resource and variable names.

Reply:
Done with latest commit, but I leave it to you to 'resolve', if you're satisfied. :)